conans/client/graph/graph_builder.py

```diff
@@ -270,14 +272,21 @@ def _resolve_replace_requires(self, node, require, profile_build, profile_host,
             break  # First match executes the alternative and finishes checking others
 
     def _create_new_node(self, node, require, graph, profile_host, profile_build, graph_lock):
-        if require.ref.version == "<host_version>":
+        require_version = str(require.ref.version)
+        if require_version == "<host_version>" or require_version.startswith("<host_version") and require_version.endswith(">"):
```

Comment:
```suggestion
        if require_version.startswith("<host_version") and require_version.endswith(">"):
```
good enough of a check?